framework/web/UrlManager.php

```diff
@@ -126,11 +126,14 @@ class UrlManager extends Component
      */
     public $ruleConfig = ['class' => 'yii\web\UrlRule'];
 
+    /**
+     * @var array internal cache to store matched rules
+     */
+    protected $ruleCache;
```

Comment:
I think we want to access $ruleCache inside subclasses. For example if we want to use parent cache logic but with some improvements (for example calculating cache hits and misses):

```php
class UrlManagerWithCacheCount extends UrlManager
{
    private $cacheHit = 0;
    private $cacheMiss = 0;
    
    protected function getUrlFromInternalCache($cacheKey, $route, $params)
    {
        if (!empty($this->ruleCache[$cacheKey])) {
            foreach ($this->ruleCache[$cacheKey] as $rule) {
                /* @var $rule UrlRule */
                if (($url = $rule->createUrl($this, $route, $params)) !== false) {
                    $this->cacheHit++;
                    return $url;
                }
            }
        } else {
            $this->ruleCache[$cacheKey] = [];
        }
        $this->cacheMiss++;
        
        return false;
    }
}
```

Otherwise we would have to make another private property. That why I see $ruleCache protected. Dont you agree?